test/unit/plugins/validate-semantic/oas3/refs.js

```diff
@@ -206,4 +206,109 @@ describe("validation plugin - semantic - oas3 refs", () => {
       return expectNoErrorsOrWarnings(spec)
     })
   })
-})
+
+  describe("response header $refs should not point to parameters", () => {
+    it("should return an error when a response header incorrectly references a local parameter component", () => {
+      const spec = {
+        openapi: "3.0.0",
+        paths: {
+          "/foo": {
+            get: {
+              responses: {
+                "200": {
+                  description: "OK",
+                  headers: {
+                    "X-MyHeader": {
+                      $ref: "#/components/parameters/MyHeader"
+                    }
+                  }
+
+                }
+              }
+            }
+          }
+        },
+        components: {
+          headers: {
+            MyHeader: {
+              $ref: "#/components/parameters/MyHeader"
+            }
+          }
```

Comment:
Suggestion: Add a dummy parameter component to avoid extra `$ref` resolution errors in this test.
```suggestion
          },
          parameters: {
            MyHeader: {}
          }
```

Reply:
Done!